ddl/scheduler.go

```diff
@@ -103,17 +110,61 @@ func NewBackfillSchedulerHandle(taskMeta []byte, d *ddl) (scheduler.Scheduler, e
 	return bh, nil
 }
 
+// UpdateStatLoop updates the row count of adding index.
+func (b *backfillSchedulerHandle) UpdateStatLoop() {
+	tk := time.Tick(time.Second * 5)
+	ser, err := infosync.GetServerInfo()
+	if err != nil {
+		logutil.BgLogger().Warn("[ddl] get server info failed", zap.Error(err))
+		return
+	}
+	path := fmt.Sprintf("distAddIndex/%d/%s:%d", b.job.ID, ser.IP, ser.Port)
+	for {
+		select {
+		case <-b.done:
+			return
+		case <-tk:
+			err := ddlutil.PutKVToEtcd(b.ctx, b.d.etcdCli, 3, path, strconv.Itoa(int(b.totalRowCnt)))
+			if err != nil {
+				logutil.BgLogger().Warn("[ddl] update row count for distributed add index failed", zap.Error(err))
+			}
+		}
+	}
+}
+
 // InitSubtaskExecEnv implements the Scheduler interface.
-func (b *backfillSchedulerHandle) InitSubtaskExecEnv(context.Context) error {
+func (b *backfillSchedulerHandle) InitSubtaskExecEnv(ctx context.Context) error {
 	logutil.BgLogger().Info("[ddl] lightning init subtask exec env")
 	d := b.d
 
-	bc, err := ingest.LitBackCtxMgr.Register(d.ctx, b.index.Unique, b.job.ID)
+	bc, err := ingest.LitBackCtxMgr.Register(ctx, b.index.Unique, b.job.ID)
 	if err != nil {
 		logutil.BgLogger().Warn("[ddl] lightning register error", zap.Error(err))
 		return err
 	}
 	b.bc = bc
+	b.ctx = ctx
+
+	ser, err := infosync.GetServerInfo()
+	if err != nil {
+		return err
+	}
+	path := fmt.Sprintf("distAddIndex/%d/%s:%d", b.job.ID, ser.IP, ser.Port)
+	response, err := d.etcdCli.Get(ctx, path)
+	if err != nil {
+		return err
+	}
+	if len(response.Kvs) > 0 {
+		cnt, err := strconv.Atoi(string(response.Kvs[0].Value))
+		if err != nil {
+			return err
+		}
+		b.totalRowCnt = int64(cnt)
```

Comment:
Is this the only place `totalRowCnt` is updated? Don't we need to update it when this subtask finished?